x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,438 @@
+  const mlTimeSeriesExplorerService = timeSeriesExplorerServiceFactory(
+    coreStart.uiSettings,
+    mlApiServices,
+    mlResultsService
+  );
+  const mlCapabilities = new MlCapabilitiesService(mlApiServices);
+  const anomalyExplorerService = new AnomalyExplorerChartsService(
+    pluginsStart.data.query.timefilter.timefilter,
+    mlApiServices,
+    mlResultsService
+  );
+
+  // Note on the following services:
+  // - `mlIndexUtils` is just instantiated here to be passed on to `mlFieldFormatService`,
+  //   but it's not being made available as part of global services. Since it's just
+  //   some stateless utils `useMlIndexUtils()` should be used from within components.
+  // - `mlFieldFormatService` is a stateful legacy service that relied on "dependency cache",
+  //   so because of its own state it needs to be made available as a global service.
+  //   In the long run we should again try to get rid of it here and make it available via
+  //   its own context or possibly without having a singleton like state at all, since the
+  //   way this manages its own state right now doesn't consider React component lifecycles.
+  const mlIndexUtils = indexServiceFactory(pluginsStart.data.dataViews);
+  const mlFieldFormatService = fieldFormatServiceFactory(mlApiServices, mlIndexUtils);
+
+  return [
+    coreStart,
+    pluginsStart as MlDependencies,
+    {
+      anomalyDetectorService,
+      anomalyExplorerService,
+      mlApiServices,
+      mlCapabilities,
+      mlFieldFormatService,
+      mlJobService,
+      mlResultsService,
+      mlTimeSeriesSearchService,
+      mlTimeSeriesExplorerService,
+      toastNotificationService,
+    },
+  ];
+};
+
+export const getSingleMetricViewerEmbeddableFactory = (
+  getStartServices: StartServicesAccessor<MlStartDependencies, MlPluginStart>
+) => {
+  const factory: ReactEmbeddableFactory<
+    SingleMetricViewerEmbeddableState,
+    SingleMetricViewerEmbeddableApi
+  > = {
+    type: ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE,
+    deserializeState: (state) => {
+      return state.rawState as SingleMetricViewerEmbeddableState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const services = await getServices(getStartServices);
+      const subscriptions = new Subscription();
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const {
+        singleMetricViewerControlsApi,
+        serializeSingleMetricViewerState,
+        singleMetricViewerComparators,
+        onSingleMetricViewerDestroy,
+      } = initializeSingleMetricViewerControls(state, titlesApi);
+
+      const dataLoading = new BehaviorSubject<boolean | undefined>(true);
+      const blockingError = new BehaviorSubject<Error | undefined>(undefined);
+      const query$ =
```

Comment:
For query and filters, you can likely use the new `fetch$` function. Additionally, to avoid the `//@ts-ignore` I'd recommend using the type guards we provide. This one would be `apiPublishesUnifiedSearch` I believe.

Reply:
Using query/filters from fetch$ now - updated in fbb2f38310a6e1b210b1f67c28d0a5958b331034